app/src/main/java/org/mozilla/rocket/content/BottomBarViewModelFactory.kt

```diff
@@ -0,0 +1,27 @@
+package org.mozilla.rocket.content
+
+import android.arch.lifecycle.ViewModel
+import android.arch.lifecycle.ViewModelProvider
+import org.mozilla.focus.Inject
+
+class BottomBarViewModelFactory private constructor() : ViewModelProvider.NewInstanceFactory() {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(BottomBarViewModel::class.java)) {
+            return BottomBarViewModel() as T
+        }
+        throw IllegalArgumentException("Unknown ViewModel class: " + modelClass.name)
+    }
+
+    companion object {
+
+        @Volatile private var INSTANCE: BottomBarViewModelFactory? = null
+
+        @JvmStatic
+        fun getInstance(): BottomBarViewModelFactory? =
```

Comment:
Just for your reference, I think singleton can be achieved with the below simpler statement while keeping thread-safe and lazy-initialize characteristic.
```companion object {
        @JvmStatic
        val instance: BottomBarViewModelFactory by lazy { BottomBarViewModelFactory() }
}```

Reply:
comment addressed